crates/rpc/tests/eip7702_tests.rs

```diff
@@ -0,0 +1,451 @@
+//! EIP-7702 delegation transaction tests for pending state.
+//!
+//! These tests verify that EIP-7702 authorization and delegation
+//! transactions work correctly in the pending/flashblocks state.
+
+use alloy_consensus::{Receipt, SignableTransaction, TxEip7702};
+use alloy_eips::{eip2718::Encodable2718, eip7702::Authorization};
+use alloy_primitives::{Address, B256, Bytes, FixedBytes, U256, address, hex};
+use alloy_provider::Provider;
+use alloy_signer::SignerSync;
+use alloy_signer_local::PrivateKeySigner;
+use base_reth_flashblocks::{Flashblock, Metadata};
+use base_reth_test_utils::{
+    fixtures::{BLOCK_INFO_TXN, BLOCK_INFO_TXN_HASH},
+    flashblocks_harness::FlashblocksHarness,
+};
+use eyre::Result;
+use op_alloy_consensus::OpDepositReceipt;
+use op_alloy_network::ReceiptResponse;
+use reth_optimism_primitives::OpReceipt;
+use rollup_boost::{ExecutionPayloadBaseV1, ExecutionPayloadFlashblockDeltaV1};
+
+// Test accounts from test_utils (Anvil deterministic keys)
+const ALICE_ADDRESS: Address = address!("f39Fd6e51aad88F6F4ce6aB8827279cffFb92266");
+const ALICE_PRIVATE_KEY: &str = "ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+
+const BOB_ADDRESS: Address = address!("70997970C51812dc3A010C7d01b50e0d17dc79C8");
+const BOB_PRIVATE_KEY: &str = "59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d";
+
+// Counter contract address (deployed in test payloads)
+const COUNTER_ADDRESS: Address = address!("e7f1725e7734ce288f8367e1bb143e90bb3f0512");
```

Comment:
instead of hardcoding, make this part of the `TestSetup`. The address will change if new transactions were added at some point before the contract deployment as the nonce change will cause the contract address to change.

You can take a look at https://github.com/madisoncarter1234/node-reth/blob/4b7efcf0b8ddbfecf5d62c806976d8f534d34f86/crates/rpc/tests/eth_call_erc20.rs#L37 for reference